### DISCUSSION TITLE
Slowness effect doesn't slow down the player

### DISCUSSION THREAD

**Discussion by NotThatRqd** [Minestom/Minestom#1946] [Help] [answered]
I'm running this code and on the client you can see that you have slowness, however it does not actually affect the speed you move. If there was some sort of setMovementSpeed that would be better too

```kt
player.addEffect(Potion(PotionEffect.SLOWNESS, 0, 60 * 20))
```

**Comment by NotThatRqd** [comment]
It's mentioned in #1547 that effects for players are client side, how do I make this happen?

**Comment by GoldenStack** [comment, marked as answer]
I believe you are intended to add an attribute to the player that decreases movement speed at the desired level. If you give yourself slowness on a Notchian server and then check your attributes this should be reflected in that.

**Reply by NotThatRqd** [reply]
Thx, this worked
```kt
p.getAttribute(Attribute.MOVEMENT_SPEED).addModifier(AttributeModifier("slowdown", -0.20, AttributeOperation.MULTIPLY_TOTAL))

```
However I want to make the player not able to sprint, is there any way to maybe give them hunger to stop them from sprinting?

**Reply by NotThatRqd** [reply]
Also with the attribute command you can set the base, can you do that in minestom?
edit: yes with .setBaseValue()

**Reply by GoldenStack** [reply]
Yeah you can stop them from sprinting by setting their hunger to 3 bars

Also if you could mark it as answered that'd be great

**Reply by NotThatRqd** [reply]
I can't figure out how to set hunger, apparently you can't at all in java edition without hunger effect but the effect doesn't do anything on the client side :/

**Reply by emortaldev** [reply]
`player.setFood()`

**Reply by NotThatRqd** [reply]
thank you
